equinox/nn/_composed.py

```diff
@@ -136,32 +139,58 @@ class Sequential(Module):
         Activation functions can be added by wrapping them in [`equinox.nn.Lambda`][].
     """
 
-    layers: tuple[_SequentialLayer, ...]
+    layers: tuple
 
-    def __init__(self, layers: Sequence[_SequentialLayer]):
+    def __init__(self, layers: Sequence[Module]):
```

Comment:
`Module` isn't quite accurate here -- we can put non-modules here too. I think `Callable` should be used here. (`from collections.abc import Callable`)

Reply:
Also, now that `_SequentialLayer` is unused, can it be deleted?